plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -3524,6 +3524,27 @@ eosio::chain::backing_store_type read_only::get_backing_store() const {
    return kv_database.get_backing_store();
 }
 
+read_only::get_all_accounts_result
+read_only::get_all_accounts( const get_all_accounts_params& params ) const
+{
+   get_all_accounts_result result;
+
+   using acct_obj_idx_type = chainbase::get_index_type<chain::account_object>::type;
+   const auto& accts = db.db().get_index<acct_obj_idx_type >().indices().get<chain::by_name>();
+
+   uint32_t i = 0;
+   uint32_t start_idx = params.page * params.page_size;
+   for (auto& a : accts) {
```

Comment:
Is it possible to directly index `accts` without doing `if (i >= start_idx)`?

Reply:
This should use the "more" pattern that other chain api calls use including the hard-coded 10ms time limit.

Reply:
@linhuang-blockone I'm now using std::advance().  This should be constant time if iterator is random access, which I'm not sure it is.   If not however, it will be no worse than previous implementation.

@heifner Implemented in new version.

Reply:
Latest update uses lower_bound and upper_bound, so clients can search based on range.